src/code_fee30_len_2d60.c

```diff
@@ -18,9 +18,26 @@ INCLUDE_ASM("code_fee30_len_2d60", spr_transform_point);
 
 INCLUDE_ASM("code_fee30_len_2d60", spr_draw_component);
 
-INCLUDE_ASM("code_fee30_len_2d60", _spr_sign_extend_12bit);
-
-INCLUDE_ASM("code_fee30_len_2d60", _spr_sign_extend_16bit);
+s32 _spr_sign_extend_12bit(u16 val) {
+    s32 temp = val & 0xFFF;
+
+    if ((temp & 0x800)) {
+        return temp | -0x1000;
```

Comment:
``| ~0x1001`` I believe is equivalent and should probably be used

Reply:
I think since -0x1000 is a rounder number it's easier to conceptualize than ~0x1001